packages/headless/src/controllers/recent-queries-list/headless-recent-queries-list.ts

```diff
@@ -199,6 +205,9 @@ export function buildRecentQueriesList(
         prepareForSearchWithQuery({
           q: this.state.queries[index],
           clearFilters: registrationOptions.clearFilters,
+          ...(isBoolean(engine.state.query?.enableQuerySyntax) && {
```

Comment:
This prompted me to make a comment in an old TypeScript that unfortunately got rejected a while back (https://github.com/microsoft/TypeScript/issues/32495#issuecomment-2065249303) regarding the fact that TypeScript doesn't check for excess properties in spread object literals, which I believe is problematic.

I don't have an efficient workaround to suggest, though.

Reply:
I've been reading on this and I think this issue has most of the details:
https://github.com/microsoft/TypeScript/issues/13444
There intended solution to this is to implement Exact types:
https://github.com/microsoft/TypeScript/issues/12936

I know it does not prevent the underlying issue from happening, but what about doing something like this instead? 

```typescript

      const queryOptions: Parameters<typeof prepareForSearchWithQuery>[0] = {
        q: this.state.queries[index],
        clearFilters: registrationOptions.clearFilters,
      };

      if (isBoolean(engine.state.query?.enableQuerySyntax)) {
        queryOptions.enableQuerySyntax = engine.state.query.enableQuerySyntax;
      }

      dispatch(prepareForSearchWithQuery(queryOptions));
```

Reply:
Forgot to mention that there's also discussions about changing the error message to make it more clear that excess property errors are more of a warning than an actual error 
https://github.com/microsoft/TypeScript/pull/55152#issuecomment-1658831938